ethers-solc/src/config.rs

```diff
@@ -468,7 +470,12 @@ pub trait ArtifactOutput {
     type Artifact: Artifact + DeserializeOwned;
 
     /// Handle the compiler output.
-    fn on_output(output: &CompilerOutput, layout: &ProjectPathsConfig) -> Result<()>;
+    fn on_output(
+        output: &CompilerOutput,
+        version: String,
```

Comment:
```suggestion
        version: &str,
```
prefer references over taking ownership. then in any `BTreeMap<_>::get` call down the road you can just use `.to_owned`